### DISCUSSION TITLE
useGLTF([urls]). Is order of returned objects guaranteed?

### DISCUSSION THREAD

**Discussion by mordv** [pmndrs/drei#606] [Q&A] [answered]
```const loaded = useGLTF(urls);```
Is it guaranteed that the order of `loaded` will be the same as the order of `urls`?

How else can I know which GLTF related to which url? I see `parser.options.path`, but it trimmed my query params, so I can't distinguish one from another.

**Comment by joshuaellis** [comment, marked as answer]
yes it should be because we're mapping through the arrays we're not splitting and applying race conditions.

**Reply by drcmda** [reply]
correct, it's 100% guaranteed. it's using Promise.all https://github.com/pmndrs/react-three-fiber/blob/master/packages/fiber/src/core/hooks.ts#L75-L84
